### ISSUE TITLE
Environmental Variables

### ISSUE THREAD

**Issue by kellrott** [ga4gh/task-execution-schemas#19]
Should we add in a mechanism to define environmental variables for a docker execution?

This was already mentioned in https://github.com/rabix/bunny/issues/75 by @simonovic86

**Comment by buchanae** [comment]
I'm split on this, so I'll just list the reasons for or against I can think of.

1. FOR: It wouldn't be that difficult to add to the schema (map<String, String>) and server (write an extra bash script, or extra docker run flags) and it would remove duplicate code in each runner and tool that was doing this.

2. AGAINST: I can't think of any use case that *requires* environment variables, and couldn't be achieved otherwise by an input file (bash script that contains environment variables).

3. FOR: Environment variables are a ubiquitous concept. They're in every programming language, every OS, every cluster engine, every container runner, etc.

4. AGAINST: Environment variables are a secondary input interface to a tool, and I think it's better to have one than two.

5. FOR: Runners are going to workaround a lack of environment variable support when they need it, probably by writing a shell script that defines the variables and using that as the container entrypoint. But, that overrides the tool's entrypoint, which is maybe a bad practice. If a tool defines an entrypoint, maybe we should respect that.

**Comment by geoffjentry** [comment]
@buchanae Our experience w/ cromwell has been that have wound up needing to write a wrapping shell script for our backends anyways, for one reason or another. Since it's going to be a lot more likely to see TES as a front end for backends as pposed to truly adopted by e.g. cloud providers, I'd imagine that TES implementations for e.g. JES/Batch/etc would also need to use wrapper scripts.

tl;dr avoiding a wrapping script is likely to not matterm uch in the long run.

**Comment by kellrott** [comment]
I think the idea that we'll always need config/runner wrapper shell scripts was one of the motivators for https://github.com/ga4gh/task-execution-schemas/issues/11 That way one could easily pass in a shell script, without having to push it to the object store first. Surprisingly people were mixed on the feature suggestion as well.

**Comment by geoffjentry** [comment]
@kellrott I definitely didn't realize that - I confess that I still have trouble at times thinking of TES as something that e.g. Cromwell is talking to vs something a human is interfacing with. In that case I believe I just made an argument in favor of #11 above ;)
